admin-dev/themes/new-theme/js/pages/product/components/categories/index.js

```diff
@@ -36,30 +36,34 @@ const {$} = window;
 
 const ProductCategoryMap = ProductMap.categories;
 
+//@todo: filename is index.js but its a class CategoriesManager. Shouldn't we keep the filename categories-manager.js?
 export default class CategoriesManager {
```

Comment:
This component should be renamed `CategoriesTreeSelector` or something similar

But you will have a new CategoriesManager that handles the "platform" between this component and the collection of tags in the form, it's also the manager which listens for click event on button and then call this component